PKCS12.xs

```diff
@@ -13,6 +13,8 @@
 #include <openssl/ssl.h>
 #if OPENSSL_VERSION_NUMBER >= 0x30000000L
 #include <openssl/provider.h>
+  OSSL_PROVIDER *legacy = NULL;
```

Comment:
This makes it global - should look at whether it should just be part of the object - as in legacy_loaded